File: ios/Runner/Repository/Contract/SongRepositoryContract.swift
//
//  SongRepositoryContract.swift
//  Runner
//
//  Created by t k on 2024/08/12.
//

protocol SongRepositoryContract {
    func findAll() -> [SongModelContract]
    
    func findBySongId(songId: SongId) -> SongModelContract?
    
    func findBySongIds(songIds: [SongId]) -> [SongModelContract]
    
    func findByAlbumId(albumId: AlbumId) -> [SongModelContract]
    
    func findByAlbumIds(albumIds: [AlbumId]) -> [SongModelContract]
}

File: android/app/src/main/kotlin/dev/tcode/thinmpf/repository/contract/SongRepositoryContract.kt
package dev.tcode.thinmpf.repository.contract

import dev.tcode.thinmpf.model.contract.SongModelContract
import dev.tcode.thinmpf.model.valueObject.AlbumId
import dev.tcode.thinmpf.model.valueObject.ArtistId
import dev.tcode.thinmpf.model.valueObject.SongId

interface SongRepositoryContract {
    fun findAll(): List<SongModelContract>

    fun findById(songId: SongId): SongModelContract?

    fun findByIds(songIds: List<SongId>): List<SongModelContract>

    fun findByAlbumId(albumId: AlbumId): List<SongModelContract>

    fun findByArtistId(artistId: ArtistId): List<SongModelContract>
}